### DISCUSSION TITLE
SOC performance issues after upgrading to 2.3.140

### DISCUSSION THREAD

**Discussion by df-sec** [Security-Onion-Solutions/securityonion#8496] [Unsupported Versions] [answered]
After upgrading our on-prem standalone node + one IDH to 2.3.140 I sometimes get the following message when logging into the SOC:

 `Disconnected from manager`

This message mostly disappeared when I go to another menue like Alerts (all alerts are displayed correctly) but when I click on "Drilldown" or "Quick Drilldown" I have to wait at least 1-2 minutes before I get any result. The same happens when acknowledging an alert or if I set any filter.

What I have tested so far:
- so-status -> OK
- GRID status -> OK
- tested with different Browsers (Edge, Chrome, Firefox) -> same result
- I've disabled all playbooks -> same result
- Kibana -> seems to works fast and fine
- SOC log -> no errors, just "Handled request" messages with statusCode=200 during the loading time
- host restarted -> same result
- elasticsearch log -> shows the following error:
 ```
[WARN ][org.elasticsearch.action.search.TransportClearScrollAction] Clear SC failed on node[{redacted}{...}{...}{redacted}{redacted}{redacted:9300}{cdfhilmrstw}{xpack.installed=true, box_type=hot}]
org.elasticsearch.transport.RemoteTransportException: [redacted][172.17.0.12:9300][indices:data/read/search[free_context/scroll]]
Caused by: org.elasticsearch.common.breaker.CircuitBreakingException: [parent] Data too large, data for [indices:data/read/search[free_context/scroll]] would be [10999440826/10.2gb], which is larger than the limit of [10885896601/10.1gb], real usage: [10999440760/10.2gb], new bytes reserved: [66/66b], usages [inflight_requests=224764/219.4kb, request=0/0b, fielddata=487873/476.4kb, eql_sequence=0/0b, model_inference=0/0b]
```

My `/opt/so/saltstack/local/pillar/minions/$minion.sls` (was already configured this way before the problem occurred):
```
elasticsearch:
  es_port: 9200
  esclustername: '{{ grains.host }}'
  esheap: 12288m
  log_size_limit: 265
  mainint: ens192
  mainip: redacted
  node_route_type: hot
  node_type: ''
...
```

What else did I miss?

Thanks in advance!

**Comment by dougburks** [comment, marked as answer]
> After upgrading our on-prem standalone node + one IDH to 2.3.140 I sometimes get the following message when logging into the SOC:
Disconnected from manager

Are there any firewall or other network devices between your workstation and your manager that could be interfering with the traffic?

> elasticsearch log -> shows the following error:
[WARN ][org.elasticsearch.action.search.TransportClearScrollAction] Clear SC failed on node[{redacted}{...}{...}{redacted}{redacted}{redacted:9300}{cdfhilmrstw}{xpack.installed=true, box_type=hot}]
org.elasticsearch.transport.RemoteTransportException: [redacted][172.17.0.12:9300][indices:data/read/search[free_context/scroll]]
Caused by: org.elasticsearch.common.breaker.CircuitBreakingException: [parent] Data too large, data for [indices:data/read/search[free_context/scroll]] would be [10999440826/10.2gb], which is larger than the limit of [10885896601/10.1gb], real usage: [10999440760/10.2gb], new bytes reserved: [66/66b], usages [inflight_requests=224764/219.4kb, request=0/0b, fielddata=487873/476.4kb, eql_sequence=0/0b, model_inference=0/0b]

Have you tried increasing your Elasticsearch heap?
https://docs.securityonion.net/en/2.3/elasticsearch.html#heap-size

**Comment by df-sec** [comment]
Hello Doug,

Thanks for your reply.

> Are there any firewall or other network devices between your workstation and your manager that could be interfering with the traffic?

No, my browser can communicate "directly" with the SOC (no proxy, reverse proxy, etc. between browser and SOC) BUT I forgot to check my AV which decrypts and scans TLS traffic. No event was generated but I had to create an exception for TLS decryption. Now it seems to be fine.

> Have you tried increasing your Elasticsearch heap?
https://docs.securityonion.net/en/2.3/elasticsearch.html#heap-size

Yes I already had set `esheap: 12288m` and rebooted the system as I mentioned. So 12GB is greater than the 10,1GB limit which can be seen in the log. Do I have to set `Xms` and `Xmx` manually?

**Reply by dougburks** [reply]
> Do I have to set Xms and Xmx manually?

Setting `esheap` should automatically set `Xms` and `Xmx` to the same value.
